File: src/main/java/xyz/chalky/taboo/commands/BookmarkContextMenuMessageCommand.java
package xyz.chalky.taboo.commands;

import net.dv8tion.jda.api.entities.Message;
import net.dv8tion.jda.api.events.interaction.command.MessageContextInteractionEvent;
import net.dv8tion.jda.api.interactions.commands.build.Commands;
import xyz.chalky.taboo.backend.MessageContextCommand;

public class BookmarkContextMenuMessageCommand extends MessageContextCommand {

    public BookmarkContextMenuMessageCommand() {
        setCommandData(Commands.message("Bookmark"));
    }

    @Override
    public void executeCommand(MessageContextInteractionEvent event) {
        Message target = event.getTarget();
        event.reply("Bookmarking message: " + target.getContentDisplay()).queue();
    }

}

File: src/main/kotlin/xyz/chalky/taboo/commands/context/BookmarkContextMenuMessageCommand.kt
package xyz.chalky.taboo.commands.context

import xyz.chalky.taboo.backend.MessageContextCommand
import net.dv8tion.jda.api.events.interaction.command.MessageContextInteractionEvent
import net.dv8tion.jda.api.interactions.commands.build.Commands

class BookmarkContextMenuMessageCommand : MessageContextCommand() {

    init {
        setCommandData(Commands.message("Bookmark"))
    }

    override fun executeCommand(event: MessageContextInteractionEvent) {
        val target = event.target
        event.reply("Bookmarking message: " + target.contentDisplay).queue()
    }

}